default/AI/DiplomaticCorp.py

```diff
@@ -0,0 +1,78 @@
+"""Handle diplomatic messages and response determination."""
+
+import random
+
+import freeOrionAIInterface as fo  # pylint: disable=import-error
+import FreeOrionAI as foAI
+import AIstate
+from freeorion_tools import UserString, UserStringList, chat_on_error, print_error
+
+
+@chat_on_error
+def handle_diplomatic_message(message):  # pylint: disable=invalid-name
+    """Handle a diplomatic message update from the server,
+    such as if another player declares war, accepts peace, or cancels a proposed peace treaty.
+    :param message:
+    :return:
+    """
+    print "Received diplomatic %s message from empire %s to empire %s" % (message.type, message.sender, message.recipient)
+    print "my empire id: %s" % fo.empireID()
+    if message.recipient != fo.empireID():
+        return
+    reply_sender = message.recipient
+    reply_recipient = message.sender
+    if message.type == fo.diplomaticMessageType.peaceProposal:
+        foAI.foAIstate.log_peace_request(message.sender, message.recipient)
+        proposal_sender_player = fo.empirePlayerID(message.sender)
+        suffix = "MILD" if foAI.foAIstate.aggression <= fo.aggression.typical else "HARSH"
+        possible_acknowledgments = UserStringList("AI_PEACE_PROPOSAL_ACKNOWLEDGEMENTS_" + suffix + "_LIST")
+        acknowledgement = random.choice(possible_acknowledgments)
+        print "Acknowledging proposal with initial message (from %d choices): '%s'" % \
+              (len(possible_acknowledgments), acknowledgement)
+        fo.sendChatMessage(proposal_sender_player, acknowledgement)
+        attitude = evaluate_diplomatic_attitude(message.sender)
+        if attitude > 0:
+            reply_text = random.choice(UserStringList("AI_PEACE_PROPOSAL_RESPONSES_YES_" + suffix + "_LIST"))
+            diplo_reply = fo.diplomaticMessage(reply_sender, reply_recipient, fo.diplomaticMessageType.acceptProposal)
+        else:
+            reply_text = random.choice(UserStringList("AI_PEACE_PROPOSAL_RESPONSES_NO_" + suffix + "_LIST"))
+            diplo_reply = None
+        fo.sendChatMessage(proposal_sender_player, reply_text)
+        print "sending chat to player %d of empire %d, message body: '%s'" % \
+              (proposal_sender_player, reply_recipient, reply_text)
+        if diplo_reply:
+            print "Sending diplomatic message to empire %s of type %s" % (reply_recipient, diplo_reply.type)
+            fo.sendDiplomaticMessage(diplo_reply)
+    elif message.type == fo.diplomaticMessageType.warDeclaration:
+        foAI.foAIstate.log_peace_request(message.sender, message.recipient)
```

Comment:
Please be less cryptic.  If your suggestion is that the AI sometimes ask for peace, then yes as has been discussed in the forums that is part of the plan.  But that decision is far more complex than what I am trying to accomplish here.